### PR TITLE
Trigger visit events

### PR DESCRIPTION
**Pull request by mackella**
# What

Added event triggers for join-visit and leave-visit.

# Why

To provide source data for deriving usage metrics.

# Screenshots

N/A

# Notes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ Please take a look at our separate [Contributing Guide](./CONTRIBUTING.md).
 - **Joshua-Luke Bevan** - Software Engineer at [Made Tech](https://www.madetech.com) (joshua.bevan@madetech.com)
 - **Stephen Thomson** - Senior Software Engineer at [Made Tech](https://www.madetech.com) (stephen.thomson@madetech.com)
 - **Neil Kidd** - Lead Software Engineer at [Made Tech](https://www.madetech.com) (neil.kidd@madetech.com)
+- **Stu Mackellar** - Lead Software Engineer at [Made Tech](https://www.madetech.com) (stu.mackellar@madettech.com)
```
Comment (chubberlisk) [inline README.md:76]: 🎉 

```suggestion
- **Stu Mackellar** - Lead Software Engineer at [Made Tech](https://www.madetech.com) (stu.mackellar@madetech.com)
```

File: pages/api/capture-event.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default withContainer(
       return;
     }
 
-    if (!body.visitId || isNaN(body.visitId)) {
+    if (body.visitId === null || body.visitId === "" || isNaN(body.visitId)) {
```
Comment (jivdhaliwal) [inline pages/api/capture-event.js:31]: Would this cover visitId being undefined?

Comment (jivdhaliwal) [inline pages/api/capture-event.js:31]: Could you clarify why we would want to accept 0 as a visitId ? (based on the message in a8ce8e9d7439c).

If we want to explicitly allow 0, have we considered including that in the statement?

(e.g. `if ((!body.visitId && body.visitId !== 0) || isNaN(body.visitId))`

This should also cover an undefined visitId

Comment (chubberlisk) [reply pages/api/capture-event.js:31]: I think at the time we reckoned 0 was a valid `visitId`. Now I'm wondering whether it is, looking at the DB?

Comment (jivdhaliwal) [reply pages/api/capture-event.js:31]: The default minimum value is 1 so we should never had visits with an ID of 0 (https://www.postgresql.org/docs/6.4/sql-createsequence.htm)

Comment (chubberlisk) [reply pages/api/capture-event.js:31]: Awesome, I'll drop the commit :)

Comment (chubberlisk) [reply pages/api/capture-event.js]: Yush, definitely now that we've dropped worrying about a 0 visitId.

**Pull request merged**